smbclient/shutil.py

```diff
@@ -0,0 +1,306 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2019, Jordan Borean (@jborean93) <jborean93@gmail.com>
+# MIT License (see LICENSE or https://opensource.org/licenses/MIT)
+from __future__ import unicode_literals
+import errno
+
+from ntpath import join, basename, normcase, abspath, normpath, splitdrive
+
+import sys
+import stat as py_stat
+from os import error as os_error
+
+from smbclient._io import ioctl_request, SMBFileTransaction, SMBRawIO
+from smbclient._os import remove, rmdir, listdir, lstat, stat, utime, open_file, makedirs, readlink, symlink
+from smbclient.path import islink, isdir
+from smbprotocol.exceptions import SMBOSError
+from smbprotocol.ioctl import SMB2SrvCopyChunk, IOCTLFlags, SMB2SrvCopyChunkResponse, CtlCode, SMB2SrvCopyChunkCopy, \
+    SMB2SrvRequestResumeKey
+from smbprotocol.open import FilePipePrinterAccessMask, CreateOptions, ShareAccess
+
+CHUNK_SIZE = 1 * 1024 * 1024  # maximum chunksize allowed by smb
```

Comment:
Where did you get this value from? SMB has a max packet size of around 65KB but newer dialects can go even larger than this.

Reply:
This has nothing to do with the smb packet size (3.3.3 in the attached document). If I read this correctly, the widely used default is 1M per chunk max and max 16MB in chunks = 16 chunks of 1M. The 16 is hidden, I should extract it as variable as well.
[[MS-SMB].pdf](https://github.com/jborean93/smbprotocol/files/3833275/MS-SMB.pdf)